model/xgboost/dffml_model_xgboost/xdgregressor.py

```diff
@@ -0,0 +1,194 @@
+# Co-authored-by: John Andersen <johnandersenpdx@gmail.com>
+# Co-authored-by: Soren Andersen <sorenpdx@gmail.com>
+import pathlib
+from typing import AsyncIterator
+import importlib
+
+import joblib
+import pandas as pd
+from sklearn.metrics import r2_score
+import xgboost as xgb
+
+from dffml.base import config, field
+from dffml.model.accuracy import Accuracy
+from dffml.model.model import SimpleModel, ModelNotTrained
+from dffml.feature.feature import Feature, Features
+from dffml.source.source import Sources
+from dffml.util.entrypoint import entrypoint
+from dffml.record import Record
+
+
+@config
+class XDGRegressorModelConfig:
+    directory: pathlib.Path = field("Directory where model should be saved")
+    features: Features = field("Features on which we train the model")
+    predict: Feature = field("Value to be predicted")
+    learning_rate: float = field("Learning rate to train with", default=0.05)
+    n_estimators: int = field(
+        "Number of graident boosted trees. Equivalent to the number of boosting rounds",
+        default=1000,
+    )
+    max_depth: int = field("Maximium tree depth for base learners", default=6)
+    subsample: float = field(
+        "Subsample ratio of the training instance", default=1
+    )
+    gamma: float = field(
+        "Minimium loss reduction required to make a furthre partition on a leaf node",
+        default=0,
+    )
+    n_jobs: int = field(
+        "Number of parallel threads used to run xgboost", default=-1
+    )
+    colsample_bytree: float = field(
+        "Subsample ratio of columns when constructing each tree", default=1
+    )
+    booster: str = field(
+        "Specify which booster to use: gbtree, gblinear or dart",
+        default="gbtree",
+    )
+    min_child_weight: float = field(
+        "Minimum sum of instance weight(hessian) needed in a child", default=0
+    )
+    reg_lambda: float = field(
+        "L2 regularization term on weights. Increasing this value will make model more conservative",
+        default=1,
+    )
+    reg_alpha: float = field(
+        "L1 regularization term on weights. Increasing this value will make model more conservative",
+        default=0,
+    )
+
+
+@entrypoint("xdgregressor")
+class XDGRegressorModel(SimpleModel):
+    """
+    Model using xgboost to perform regression prediction via gradient boosted trees
+    XGBoost is a the leading software library for working with standard tabular data (the type of data you store in Pandas DataFrames,
+    as opposed to more exotic types of data like images and videos). With careful parameter tuning, you can train highly accurate models.
+    Parameters for xgboost:
+        n_estimators = 100-1000 range,
+        learning_rate - In general, a small learning rate and large number of estimators will yield more accurate XGBoost models
+            e.g. learning_rate=0.1
+        n_jobs - specify number of cores to run in parallel
```

Comment:
We avoid talking about parameters in the model's docstring, since they'll be shown on the plugins page generated from the config anyway

For example, the `Args` section here: https://intel.github.io/dffml/master/plugins/dffml_model.html#dffml-model-tensorflow

We should include the Python example you've written here too.

```suggestion

    Examples
    --------

    Python usage

    .. literalinclude:: /../model/xgboost/examples/diabetesregression.py

    Output

    .. code-block::

        <Replace with output of running the example>
```

Reply:
I have added the example - however even after a lot of fiddling I can't get the right parameters to get a very good result on the diabetes dataset. So it has quite poor accuracy. Is this okay? Do you have any tips for finding better parameters?